code-challenges.js

```diff
@@ -23,19 +44,71 @@ var middleLetters1 = "hello"
 var middleLetters2 = "rhinoceros"
 // Expected output: “oc”
 
+ const extractMiddle = (str) => {
+
+        var position;
+        var length;
+
+        if(str.length % 2 == 1) {
+            position = str.length / 2;
+            length = 1;
+        } else {
+            position = str.length / 2 - 1;
+            length = 2;
+        }
+
+        return str.substring(position, position + length)
+    }
+
+    console.log(extractMiddle(middleLetters1))
+    console.log(extractMiddle(middleLetters2))
+    // To be completely honest, I don't quite understand the logic behind this one. I had to google it. I know it works, but I don't know how it works.
```

Comment:
that's ok! let me know if assessment review helped or if you'd like to walk through it together.